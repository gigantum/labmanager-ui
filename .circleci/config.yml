version: 2
jobs:
  build:
    docker:
      - image: gigantum/circleci-service-client:992bcbab-2018-07-09
    steps:
      - checkout
      - setup_remote_docker
      - run:
          name: Initialize submodules
          command: |
            git submodule update --init --recursive
      - run:
          name: Build frontend production code
          command: |
            npm install
            npm install relay
            npm run relay
            unset CI && npm run build
      - run:
          name: Initialize submodules
          command: |
            jest dashboard -u
