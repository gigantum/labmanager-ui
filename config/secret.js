const secrets = {
  accessToken: 'eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiIsImtpZCI6Ik56a3dRekV5TVRORFJVTXpRakkzUmtRMlFrVkVRME0xT0RCRE5qRXhNalJHT1VOQ1JEQkJRZyJ9.eyJpc3MiOiJodHRwczovL2dpZ2FudHVtLmF1dGgwLmNvbS8iLCJzdWIiOiJhdXRoMHw1OWU4YzgyYmMzMGEzODA1M2FiNTM2YjMiLCJhdWQiOlsiYXBpLmdpZ2FudHVtLmlvIiwiaHR0cHM6Ly9naWdhbnR1bS5hdXRoMC5jb20vdXNlcmluZm8iXSwiaWF0IjoxNTI3MDg4Mzg1LCJleHAiOjE1MjcxNzQzODUsImF6cCI6IjBBam84T3Y2UXJpYWZvMHhWcFN5ZWpXeTFwd05XZGhrIiwic2NvcGUiOiJvcGVuaWQgcHJvZmlsZSBlbWFpbCJ9.QHxIdCtoHr1DgAmyYhxQ70J2p0j0HFBDAuCPPT_Vr6yVFsyJNMvYLivedvgM7HJ2WcSgc84ZEeaG95SNkTl21wA4GnpP2vLJc82DecFm-iDM5ip-6fvW9-Gr0HTRIldfAe-qtiLy70DMbBuQW7zqEMAs8FHIn8zqPpG0AD9toGBqXWlX-wCCIVobG9ACHrzWp5HpQmnxtN1tXEFIdfMv6xdFM5aKhcNLf8WEWRxueOdXvD697tB921bxWpxIFqy-pKiAsHK5hMc0JT9e1DLwqnsu6Qt2TzA69hI4mTyhfRZiemrRY82yUggCMyBjC1YjtZMjKem2g3CHmnFR-P5bGQ',
  'username': 'front-end-test'
}

export default secrets
