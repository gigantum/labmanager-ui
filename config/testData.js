import developerConfig

export const testData = {
  createLabbookMutation:{
    labbookName: 'ui-test-automated',
    'owner': 'gigantum_test'
  }
}
