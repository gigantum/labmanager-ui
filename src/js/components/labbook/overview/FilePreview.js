// vendor
import React, { Component } from 'react';
import {
  QueryRenderer,
  graphql,
} from 'react-relay';
import { Link } from 'react-router-dom';
// components
import Loader from 'Components/shared/Loader';
import FileCard from './FileCard';
import FileEmpty from './FileEmpty';
import ToolTip from 'Components/shared/ToolTip';
// utilites
import environment from 'JS/createRelayEnvironment';
// store
import store from 'JS/redux/store';
// assets
import './FilePreview.scss';

const filePreviewQuery = graphql`query FilePreviewQuery($name: String!, $owner: String!, $first: Int!){
  labbook(name: $name, owner: $owner){
    code{
      favorites(first: $first){
        edges{
          node{
            id
            owner
            name
            section
            key
            description
            isDir
          }
        }
      }
    }
    input{
      favorites(first: $first){
        edges{
          node{
            id
            owner
            name
            section
            key
            description
            isDir
          }
        }
      }
    }
    output{
      favorites(first: $first){
        edges{
          node{
            id
            owner
            name
            section
            key
            description
            isDir
          }
        }
      }
    }
  }
}
`;

export default class FilePreview extends Component {
  render() {
    const { owner, labbookName } = store.getState().routes;
    return (
      <QueryRenderer
        variables={{
        name: labbookName,
        owner,
        first: 3,
      }}
        query={filePreviewQuery}
        environment={environment}
        render={({ error, props }) => {
        if (props) {
          return (
            <div className="FilePreview">
              <div className="FilePreview__section">
                <div className="FilePreview__container">
                  <h5>Code &nbsp;&nbsp; <ToolTip section="codeOverview" /></h5>
                  <Link
                    onClick={this.props.scrollToTop}
                    to={{ pathname: `../../../../projects/${owner}/${labbookName}/code` }}
                    replace
                  >
                    Code Details >
                  </Link>
                </div>
                <p>Favorite Code Files</p>
                <div className="FilePreview__list grid">
                  {
                    props.labbook.code.favorites && props.labbook.code.favorites.edges.length ?
                    props.labbook.code.favorites.edges.map(edge => <FileCard key={edge.node.id} edge={edge} />) :
                    <FileEmpty
                      section="code"
                      mainText="This Project has No Code Favorites"
                      subText="View Project Code Details"
                    />
                  }
                </div>

              </div>
              <div className="FilePreview__section">
                <div className="FilePreview__container">
                  <h5>Input Data<ToolTip section="inputDataOverview" /></h5>
                  <Link
                    onClick={this.props.scrollToTop}
                    to={{ pathname: `../../../../projects/${owner}/${labbookName}/inputData` }}
                    replace
                  >
                    Input Data Details >
                  </Link>
                </div>
                <p>Favorite Input Files</p>
                <div className="FilePreview__list grid">
                  {
                    props.labbook.input.favorites &&
                    props.labbook.input.favorites.edges.length ? props.labbook.input.favorites.edges.map(edge => <FileCard key={edge.node.id} edge={edge} />) :
                    <FileEmpty
                      section="inputData"
                      mainText="This Project has No Input Favorites"
                      subText="View Project Input Data Details"
                    />
                  }
                </div>
              </div>
              <div className="FilePreview__section">
                <div className="FilePreview__container">
                  <h5>Output Data<ToolTip section="outputDataOverview" /></h5>
                  <Link
                    onClick={this.props.scrollToTop}
                    to={{ pathname: `../../../../projects/${owner}/${labbookName}/outputData` }}
                    replace
                  >
                    Output Data Details >
                  </Link>
                </div>
                <p>Favorite Output Files</p>
                <div className="FilePreview__list grid">
                  {
                    props.labbook.output.favorites &&
                    props.labbook.output.favorites.edges.length ? props.labbook.output.favorites.edges.map(edge => <FileCard key={edge.node.id} edge={edge} />) :
                    <FileEmpty
                      section="outputData"
                      mainText="This Project has No Output Favorites"
                      subText="View Project Output Data Details"
                    />
                  }
                </div>
              </div>
            </div>
          );
        } else if (error) {
          return (<div>{error.message}</div>);
        }
          return (<Loader />);
      }}
      />);
  }
}
