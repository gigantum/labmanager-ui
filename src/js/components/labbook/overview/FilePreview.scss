@import '~Styles/_imports.scss';

.FilePreview__section {
  margin-top: 20px;
}

.FilePreview__container {
  align-items: center;
  @include flex(space-between, row);
  a {
    color: $dark-blue;
  }
}
