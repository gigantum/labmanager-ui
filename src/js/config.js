import tips from './tips'

const CONFIG = {
  'api': process.env.NODE_ENV,
  'navigation_items': [
    {id:'overview', name: 'Overview'},
    {id:'activity', name: 'Activity', 'fragment': '...Activity_labbook'},
    {id:'environment', name: 'Environment', 'fragment': '...Environment_labbook'},
    {id:'code', name: 'Code'},
    {id:'inputData', name: 'Input Data'},
    {id:'outputData', name: 'Output Data'}
  ],
  'modalNav': [
    {'id': 'createLabook', 'description': 'Title & Description'},
    {'id': 'selectBaseImage', 'description': 'Base Image'},
    {'id': 'selectDevelopmentEnvironment', 'description': 'Dev Environment'},
    {'id': 'addEnvironmentPackage', 'description': 'Add Dependencies'},
    {'id': 'successMessage', 'description': 'Success'}
  ],
  'months': [
    'Jan',
    'Feb',
    'Mar',
    'Apr',
    'May',
    'Jun',
    'Jul',
    'Aug',
    'Sep',
    'Oct',
    'Nov',
    'Dec'
  ],
  'fileBrowser':{
    'excludedFiles': [
      'DS_Store',
      'zip',
      'lbk',
      'pyc',
      'gitkeep'
    ]
  },

  containerStatus:{
    canEditEnvironment: (status) => {
      return (status === 'Stopped') || (status === 'Rebuild')
    }
  },
  userAPI: {
    getUsersQueryString: (userInput) => {
      const sanitizedUserInput = userInput.replace(/-/g, ' ')
      const apiURL = `https://m9eq4m3z0f.execute-api.us-east-1.amazonaws.com/prod?q=${sanitizedUserInput}*&q.options={fields: ['username^5','name']}&size=10`

      return encodeURI(apiURL)
    },
    getUserEmailQueryString: (email) => {
      const apiURL = `https://m9eq4m3z0f.execute-api.us-east-1.amazonaws.com/prod?q=${email}&q.options={fields: ['email']}&size=10`

      return encodeURI(apiURL)
    }
  },
  getToolTipText: (section) =>{
    return tips[section]
  },
  demoHostName: 'try.gigantum.com',
}

export default CONFIG
