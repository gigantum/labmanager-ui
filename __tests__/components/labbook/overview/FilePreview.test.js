
      import React from 'react'
      import renderer from 'react-test-renderer';
      import {mount} from 'enzyme'
      import FilePreview from 'Components/labbook/overview/FilePreview';

      import store from 'JS/redux/store'

      import json from './__relaydata__/FilePreview.json'

      import config from '../../../config'

      import relayTestingUtils from 'relay-testing-utils'

      const fixtures = {
        labbook: json.data.labbook
      }

      store.dispatch({
        type: 'UPDATE_ALL',
        payload: {
          labbookName: config.labbookName,
          owner: config.owner
        }
      })


      describe('Test FilePreview', () => {
        it('renders snapshot', ()=> {
          const wrapper = renderer.create(

             relayTestingUtils.relayWrap(<FilePreview {...fixtures} />, {}, json.data.labbook)

          );

          const tree = wrapper.toJSON()

          expect(tree).toMatchSnapshot()
        })


      })
