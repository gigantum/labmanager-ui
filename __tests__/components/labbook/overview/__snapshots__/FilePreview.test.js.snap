// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test FilePreview renders snapshot 1`] = `
<div
  className="Loader"
/>
`;
